src/default-config-backend.cpp

```diff
@@ -34,37 +77,26 @@ static int handle_config_updated(int fd, uint32_t mask, void *data)
     }
 
     LOGD("Reloading configuration file");
-
     char buf[INOT_BUF_SIZE] __attribute__((aligned(alignof(inotify_event))));
 
     bool should_reload = false;
-    inotify_event *event;
-
-    while (true)
+    if (read_event(fd, buf))
     {
-        auto len = read(fd, buf, INOT_BUF_SIZE);
-        if (len < 0)
+        auto event = reinterpret_cast<inotify_event*>(buf);
+        if (event->len == 0)
```

Comment:
I would expect this to be `if (((event->mask & IN_CREATE) && event->len > 0 && config_file == event->name) || event->mask & IN_MODIFY)`. If the file is created, it should give us the file name in the event. If the file existed and it was modified, we already know the file existed before we added the watch for it.